neuralop/models/base_model.py

```diff
@@ -76,7 +76,33 @@ def __new__(cls, *args, **kwargs):
         instance._init_kwargs = kwargs
 
         return instance
-    
+
+    def state_dict(self, **kwargs):
+        """
+        state_dict subclasses nn.Module.state_dict() and adds a metadata field
+        to track the model version and ensure only compatible saves are loaded.
+        """
+        print("In state dict")
+        state_dict = super().state_dict(**kwargs)
+        state_dict.update({'neuralop_metadata': self._init_kwargs})
```

Comment:
Shall we just call it `_metadata`? Perhaps we add a test to check that there isn't already a key under that name in the unlikely event that this happened.
I would also just do `state_dict['_metadata'] = self._init_kwargs` since we are only updating the values for one key.

Reply:
We can just call it `'_metadata'`, but we need to update because the PyTorch state dict doesn't support item assignment (throws an error)